src/melody/melody.go

```diff
@@ -254,72 +268,92 @@ func (m *Melody) submitFileForSeedingForm(w http.ResponseWriter, r *http.Request
 		w.Write([]byte("Error writing the file to storage"))
 		return
 	}
-	defer f.Close()
-	io.Copy(f, receivedFile)
-
-	w.WriteHeader(http.StatusOK)
-	w.Write([]byte(fileId))
-	m.AddPeerServingFile(m.address, fsi.Metadata)
-}
+	nr := io.TeeReader(receivedFile, f)
 
-func (m *Melody) submitFileForSeeding(w http.ResponseWriter, r *http.Request) {
-	query := r.URL.Query()
-	w.Header().Set("Access-Control-Allow-Origin", "*")
-	if len(query["fileid"]) == 0 {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Missing required query param 'fileid'"))
+	hash := sha256.New()
+	if _, err := io.Copy(hash, nr); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("Could not compute the hash of the uploaded file"))
+		f.Close()
 		return
 	}
 
-	fileId := query["fileid"][0]
-
-	fsi := m.LocateSeeders(fileId)
-	if fsi.Metadata.Id == "" || fsi.Metadata.Id != fileId {
+	hexhash := hex.EncodeToString(hash.Sum(nil))
+	if hexhash != fsi.Metadata.Hash {
 		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Could not find the fileId in the network"))
+		w.Write([]byte("The hash of the seeded file did not match the hash stored."))
+		f.Close()
+		os.Remove(m.getStoreDirectoryName() + fileId)
 		return
 	}
 
-	// Ideally this would be streamed to the client.
-	// However, loading in bytes in memory if ok for now for our prototype.
-	filedata, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("Could not read the file sent in the request"))
-		return
-	}
-
-	os.MkdirAll(m.getStoreDirectoryName(), os.ModePerm)
-
-	files, err := ioutil.ReadDir(m.getStoreDirectoryName())
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("Could not find the files being seeded"))
-		return
-	}
-
-	for _, f := range files {
-		if f.Name() == fileId {
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte("Already seeding file with the given ID"))
-			return
-		}
-	}
-
-	f, err := os.Create(m.getStoreDirectoryName() + fileId)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		w.Write([]byte("Error writing the file to storage"))
-		return
-	}
 	defer f.Close()
-	f.Write(filedata)
-
 	w.WriteHeader(http.StatusOK)
 	w.Write([]byte(fileId))
 	m.AddPeerServingFile(m.address, fsi.Metadata)
 }
 
+//
+// This method is for adding the file as a binary data in the request.
+//
+// func (m *Melody) submitFileForSeeding(w http.ResponseWriter, r *http.Request) {
```

Comment:
I dont think anything was using this. I had it to use binary data directly without creating forms, but dont want to keep this around if we dont need it rn.